### PR TITLE
8226933: [TEST_BUG]GTK L&F: There is no swatches or RGB tab in JColorChooser

### PR DESCRIPTION
**Pull request by kumarabhi006**
There is no Swatches tab available for GTK Look and Feel due to the different ColorChooser UI design. Updated the test instructions.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8226933](https://bugs.openjdk.org/browse/JDK-8226933): [TEST_BUG]GTK L&amp;F: There is no swatches or RGB tab in JColorChooser (**Bug** - P4)

### Reviewers
 * [Prasanta Sadhukhan](https://openjdk.org/census#psadhukhan) (@prsadhuk - **Reviewer**) Review applies to [677cb0e9](https://git.openjdk.org/jdk/pull/20867/files/677cb0e9452239dd8b026bf7abfeb96a76533161)
 * [Tejesh R](https://openjdk.org/census#tr) (@TejeshR13 - **Reviewer**) Review applies to [56bb68d9](https://git.openjdk.org/jdk/pull/20867/files/56bb68d95bd0fbb39d5f0cdece4119def1601d1f)
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20867/head:pull/20867` \
`$ git checkout pull/20867`

Update a local copy of the PR: \
`$ git checkout pull/20867` \
`$ git pull https://git.openjdk.org/jdk.git pull/20867/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20867`

View PR using the GUI difftool: \
`$ git pr show -t 20867`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20867.diff">https://git.openjdk.org/jdk/pull/20867.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20867#issuecomment-2333305306)

### CODE REVIEW

File: test/jdk/javax/swing/JColorChooser/Test4887836.java [hunk 1]
```diff
@@ -40,14 +40,16 @@ public class Test4887836 {
     public static void main(String[] args) throws Exception {
         String instructions = """
                                 If you do not see white area under the \"Swatches\" tab,
-                                then test passed, otherwise it failed.""";
+                                then test passed, otherwise it failed.
+
+                                NOTE: There is no Swatches tab for GTK Look and Feel, Press Pass.""";
```
Comment (prsadhuk) [inline test/jdk/javax/swing/JColorChooser/Test4887836.java:45]: I guess better to restrict programmatically 
"GTK".equals(UIManager.getLookAndFeel().getName())
Also, add a comment why it is not applicable..

Comment (prsadhuk) [reply test/jdk/javax/swing/JColorChooser/Test4887836.java:45]: Whatever L&F you are running on irrespective of manually or automated, it should be able to find out via getLookAndFeel..

Comment (kumarabhi006) [reply test/jdk/javax/swing/JColorChooser/Test4887836.java]: Yeah, you are right. Test updated to throw `jtreg.SkippedException` if it is run for GTK L&F.

**Review by prsadhuk — APPROVED** [review]

**Review by TejeshR13 — APPROVED** [review]
Looks good to me.

**Comment by kumarabhi006** [comment]
@prsadhuk @TejeshR13 bug4419255.java is open sourced. Fixed similar issue for ColorChooser UI in GTK L&F. 
Please re-review.

File: test/jdk/javax/swing/plaf/basic/BasicSliderUI/bug4419255.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ */
+
+public class bug4419255 {
+
+    public static void main(String[] args) throws Exception {
+
+        // ColorChooser UI design is different for GTK L&F.
+        // There is no RGB tab available for GTK L&F, skip the testing.
+        if (UIManager.getLookAndFeel().getName().contains("GTK")) {
+            System.out.println("Test skipped for " + UIManager.getLookAndFeel().getName());
```
Comment (prsadhuk) [inline test/jdk/javax/swing/plaf/basic/BasicSliderUI/bug4419255.java:46]: guess no need to enquire it again in as we know it's GTK

File: test/jdk/javax/swing/JColorChooser/Test4887836.java [hunk 3]
```diff
@@ -38,6 +40,14 @@
 public class Test4887836 {
 
     public static void main(String[] args) throws Exception {
+
+        // ColorChooser UI design is different for GTK L&F.
+        // There is no Swatches tab available for GTK L&F, skip the testing.
+        if (UIManager.getLookAndFeel().getName().contains("GTK")) {
+            System.out.println("Test skipped for " + UIManager.getLookAndFeel().getName());
```
Comment (prsadhuk) [inline test/jdk/javax/swing/JColorChooser/Test4887836.java:47]: same here

File: test/jdk/javax/swing/plaf/basic/BasicSliderUI/bug4419255.java (hunk 2 above)
Comment (kumarabhi006) [reply test/jdk/javax/swing/plaf/basic/BasicSliderUI/bug4419255.java]: I think print statement is not required, anyways it is captured in exception, so removed.

File: test/jdk/javax/swing/JColorChooser/Test4887836.java (hunk 3 above)
Comment (kumarabhi006) [reply test/jdk/javax/swing/JColorChooser/Test4887836.java]: Removed print statement here as well.

**Review by prsadhuk — APPROVED** [review]

**Comment by alisenchung** [comment]
Since these two tests are the same test would it be better to merge them into 1 test with the test instructions combined for both tests?

**Review by alisenchung — COMMENTED** [review]
Since these two tests are the same test would it be better to merge them into 1 test with the test instructions combined for both tests?

File: test/jdk/javax/swing/plaf/basic/BasicSliderUI/bug4419255.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                .columns(40)
+                .testTimeOut(5)
+                .testUI(bug4419255::createColorChooser)
+                .build()
+                .awaitAndCheck();
+    }
+
+    private static JColorChooser createColorChooser() {
+        JColorChooser chooser = new JColorChooser(Color.BLUE);
+        return chooser;
```
Comment (TejeshR13) [inline test/jdk/javax/swing/plaf/basic/BasicSliderUI/bug4419255.java:66]:
```suggestion
        return new JColorChooser(Color.BLUE);;
```

File: test/jdk/javax/swing/JColorChooser/Test4887836.java [hunk 5]
```diff
@@ -26,10 +26,12 @@
...
 import javax.swing.UIManager;
 
+import jtreg.SkippedException;
+
 /*
  * @test
  * @bug 4887836
- * @library /java/awt/regtesthelpers
+ * @library /java/awt/regtesthelpers /test/lib
  * @build PassFailJFrame
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/JColorChooser/Test4887836.java:35]: You may explicitly mention `jtreg.SkippedException` in the `@build` tag.
```suggestion
 * @build PassFailJFrame jtreg.SkippedException
```

**Review by aivanov-jdk — APPROVED** [review]

File: test/jdk/javax/swing/JColorChooser/Test4887836.java (hunk 5 above)
Comment (kumarabhi006) [reply test/jdk/javax/swing/JColorChooser/Test4887836.java:35]: Left it for now.

**Review by aivanov-jdk — APPROVED** [review]

**Pull request closed without merging**